crates/starknet-devnet/tests/common/background_devnet.rs

```diff
@@ -225,12 +225,24 @@ impl BackgroundDevnet {
     }
 
     /// Get balance at contract_address, as written in the ERC20 contract corresponding to `unit`
+    /// from latest block state
     pub async fn get_balance(
         &self,
         address: &FieldElement,
         unit: FeeUnit,
     ) -> Result<FieldElement, anyhow::Error> {
-        let params = format!("address={:#x}&unit={}", address, unit);
+        Self::get_balance_by_tag(self, address, unit, BlockTag::Latest).await
+    }
+
+    /// Get balance at contract_address, as written in the ERC20 contract corresponding to `unit`
+    /// from pending state or latest state
+    pub async fn get_balance_by_tag(
+        &self,
+        address: &FieldElement,
+        unit: FeeUnit,
+        tag: BlockTag,
+    ) -> Result<FieldElement, anyhow::Error> {
+        let params = format!("address={:#x}&unit={}&tag={}", address, unit, Self::tag_to_str(tag));
```

Comment:
Something doesn't feel right here. We now have: `get_balance`, `get_balance_by_block` and `get_balance_by_tag`. The functionality of `get_balance_by_tag` is a subset of the functionality of `get_balance_by_block`. We can rename `get_balance` to `get_latest_balance` (or `get_balance_latest`).

The confusion grows even more with the introduction of `get_contract_balance` in L262. IMO that method doesn't belong to `BackgroundDevnet`, it's a utility method that should be better documented (explaining what exactly is this balance - not the usual ERC20 balance), and used as a util function (defined outside of BackgroundDevnet).

Reply:
3ebad46f679dfbad34ca9e862352e506a5e2ea27